library/core/src/fmt/rt.rs

```diff
@@ -65,8 +66,11 @@ pub(super) enum Flag {
 }
 
 #[derive(Copy, Clone)]
-enum ArgumentType<'a> {
-    Placeholder { value: &'a Opaque, formatter: fn(&Opaque, &mut Formatter<'_>) -> Result },
+enum ArgumentType {
```

Comment:
Hm, why did you remove the lifetime here?

My plan was to add a `PhantomData<&'a ()>` to represent the borrow. After all, the `value` does have a limited lifetime and that should be reflected in this type, no?

Reply:
It's added to `Argument` below, I didn't want to add it to the `enum`, that seems unstylish.

Reply:
But this is where the pointer lives. There's a key invariant that `value` is a pointer valid for reading at some dynamic type `T` during lifetime `'a`. By removing the lifetime from ArgumentType you  can't even *state* that invariant as the lifetime is not in scope.